File: UI Automation C#/Exercise - Maintain a POM Framework/SDET_dotnet_MstestV2/Base/UnoBrowser.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Interactions;
using System;
using System.Collections.Generic;
using System.IO;
using System.Reflection;
using System.Text;

namespace SDET_dotnet_MstestV2.Base
{
    public class UnoBrowser
    {
        IWebDriver driver;
        public enum Browser
        {
            Chrome
        }

        public IWebDriver CreateBrowser(Browser browser)
        {
            //driver = new ChromeDriver(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location));
            driver = new ChromeDriver(@"C:\Users\heriberto.vasquez\Documents\WebDrivers\Chrome"); //local webdriver
            driver.Manage().Window.Maximize();
            driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(6);
            driver.Navigate().GoToUrl("https://www.unosquare.com");
            return driver;
        }

        public void Click(IWebElement element)
        {
            element.Click();
        }

        public int countElements(String xpath)
        {
            int elements = driver.FindElements(By.XPath(xpath)).Count;
            return elements;
        }

        public List<IWebElement> FindElements(String xpath)
        {
            List<IWebElement> elements = new List<IWebElement>(driver.FindElements(By.XPath(xpath)));
            return elements;
        }

        public void MoveToElement(IWebElement element)
        {
            Actions action = new Actions(driver);
            action.MoveToElement(element);
            action.Perform();
        }

    }
}

File: UI Automation Java/Exercise - Maintain a POM Framework/automation_POMFramework/src/main/java/Base/UnoBrowser.java
package Base;
import automation_TestNG.BrowserController;

import java.util.concurrent.TimeUnit;

import org.junit.Assert;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;

public class UnoBrowser extends BrowserController {
	WebDriver driver;
	@Override
	public WebDriver CreateBrowser() {
		System.setProperty("webdriver.chrome.driver", "C:\\Users\\heriberto.vasquez\\Documents\\WebDrivers\\Chrome\\chromedriver.exe");
		driver = new ChromeDriver();
		driver.manage().window().maximize();
		driver.get("https://www.amazon.com/");
		driver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);
		return driver;
	}
	
	@Override
	public WebElement findElement(WebDriver driver, By xpath) {
		return driver.findElement(xpath);
	}

	@Override
	public void SendText(WebElement element, String value) {
		element.sendKeys(value);		
	}

	@Override
	public void Click(WebElement element) {
		element.click();
	}
	
	@Override
	public void displayed(WebElement element){
		Assert.assertTrue(element.isDisplayed());
		System.out.println("ON SCREEN!!");
	}

	@Override
	public void isEquals(String str1, String str2){
		Assert.assertEquals(str1,str2);
		System.out.println("PASSED!!");
	}
	
	@Override
	public void get(String url){
		driver.get(url);
	}

}
